plugins/node/opentelemetry-instrumentation-express/src/instrumentation.ts

```diff
@@ -269,7 +269,7 @@ export class ExpressInstrumentation extends InstrumentationBase<
               req.res?.removeListener('finish', onResponseFinish);
               span.end();
             }
-            if (!(req.route && arguments[0] instanceof Error)) {
+            if (!(req.route && arguments[0] instanceof Error) && layerPath) {
```

Comment:
so we are over pop item right,

If layerPath is undefined, we don't append new record _LAYERS_STORE_PROPERTY. But when next is call, in the current implementation we still pop it.